unsupervised_da_exp.py

```diff
@@ -166,79 +238,98 @@ def train_da_model(classifier, datasets, train_ds, test_ds, num_tr_datasets,
   """
   # Starting DA
   fitting_info = mt.get_losses_and_callbacks(
-      datasets=datasets, num_tr_datasets=num_tr_datasets, prefix=prefix,
-      add_mw_callback=False, end_lr_coefficient=0.3, train_params=train_params)
+      datasets=datasets, num_tr_datasets=num_tr_datasets, prefix="DA",
+      add_mw_callback=False, end_lr_coefficient=1., train_params=train_params)
   callbacks, lr_schedule, losses, loss_weights, ckpt = fitting_info
+  # creating a DA model
+
   da_model = get_da_model(classifier, datasets, losses=losses,
-                          lr_schedule=lr_schedule, loss_weights=loss_weights,
-                          da_params=da_params)
+                          loss_weights=loss_weights,
+                          lr_schedule=lr_schedule,
+                          da_params=da_params,
+                          restore=train_params.restore > 0, ckpt_path=ckpt,
+                          model_params=model_params)
+  # evaluate the source-only accuracy
+  eval_source_only(
+      da_model, datasets=datasets, target_domain=da_params.target_domain,
+      num_layers=train_params.num_layers, results_file=results_file,
+      da_mode=da_params.da_mode, test_ds=train_ds,
+      copy_from=da_params.copy_from, num_steps=train_params.num_steps)
+
   print("Training DA model on source and target")
   # fit the DA model
+  da_model.train_classifier = True
   da_model.fit(
       train_ds.prefetch(tf.data.experimental.AUTOTUNE),
-      epochs=train_params.num_epochs, steps_per_epoch=train_params.num_steps,
-      validation_data=test_ds, callbacks=callbacks)
+      epochs=da_params.num_epochs_da, steps_per_epoch=train_params.num_steps,
+      validation_steps=train_params.num_steps,
+      validation_data=train_ds, callbacks=callbacks)
   da_model.save_weights(ckpt)
+  scores = da_model.evaluate(train_ds, steps=train_params.num_steps)
+  mt.write_scores(datasets, scores, results_file, f_mode="a+",
+                  scores_name="DA train")
+  scores = da_model.evaluate(test_ds)
+  mt.write_scores(datasets, scores, results_file, f_mode="a+",
+                  scores_name="DA test")
   return da_model
 
+
+def restore_mmv2_classifier(classifier, ds_info, model_params, ckpt_path):
```

Comment:
Make sure most functions like these have docstrings. I would also encourage type annotations, but this would take quite some time to add them throughout the source code.